packages/kbn-apm-synthtrace/src/cli/utils/synthtrace_worker.ts

```diff
@@ -27,23 +29,38 @@ const { bucketFrom, bucketTo, runOptions, esUrl, version } = workerData as Worke
 
 async function start() {
   const logger = loggerProxy;
-  const apmEsClient = getEsClient({
+  const apmEsClient = getApmEsClient({
     concurrency: runOptions.concurrency,
     target: esUrl,
     logger,
     version,
   });
 
+  const logsEsClient = getLogsEsClient({
+    concurrency: runOptions.concurrency,
+    target: esUrl,
+    logger,
+  });
+
   const file = runOptions.file;
 
+  let client: ApmSynthtraceEsClient | LogsSynthtraceEsClient = apmEsClient;
+
   const scenario = await logger.perf('get_scenario', () => getScenario({ file, logger }));
 
   logger.info(`Running scenario from ${bucketFrom.toISOString()} to ${bucketTo.toISOString()}`);
 
-  const { generate, bootstrap } = await scenario({ ...runOptions, logger });
+  const { generate, bootstrap, setClient } = await scenario({ ...runOptions, logger });
```

Comment:
Nope, can you elaborate a bit more. 
We always do this 2 step process.
1. Generate data (Happens in scenario files for CLI and similarly in helper functions in Tests)
2. We then use the respective client to run through the pipeline to transform the data and at the end bulk write it to ES.

Do you expect the scenario files to do both now, like generate events and then initialise the respective client and handle the indexing part, like we do in tests ?

This would also mean rewriting alot of stuff in the CLI folder like `live_data_upload` and `historical_data_upload` and all the scenario files. I am way aiming not to touch existing working code.